lib/mat_SDCZ.ml

```diff
@@ -158,6 +158,63 @@ let to_col_vecs mat =
     for i = 2 to n do ar.(i - 1) <- col mat i done;
     ar
 
+let of_col_vecs_list = function
+  | [] -> empty
+  | (vec :: _) as lst ->
+    let n = List.length lst in
+    let m = Array1.dim vec in
+    let mat = create m n in
+    let rec loop i = function
+      | [] -> mat
+      | vec :: t ->
+        if Array1.dim vec <> m then
+          failwith "of_col_vecs_list: vectors not of same length";
+        direct_copy ~n:m ~ofsy:1 ~incy:1 ~y:(col mat i) ~ofsx:1 ~incx:1 ~x:vec;
+        loop (i + 1) t
+    in
+    loop 1 lst
+
+let to_col_vecs_list mat =
+  let n = dim2 mat in
+  let rec loop i acc =
+    if i < 1 then acc
+    else
+      loop (i - 1) (col mat i :: acc)
+  in
+  loop n []
+
+let of_list = function
+  | [] -> empty
+  | (h :: _) as lst ->
+    let m = List.length lst in
+    let n = List.length h in
+    let mat = create m n in
+    let rec loop i = function
+      | [] -> mat
+      | h :: t ->
+        let s = List.fold_left (fun j e -> mat.{i, j} <- e; j + 1) 1 h in
+        if s <= n then
+          invalid_arg "list not long enough"
```

Comment:
Here you use `Invalid_argument` and later `Failure` for errors that in substance concern the same kind of problem.  To be coherent with the other functions, I think `Failure` should be used for both.